pynapple/process/filtering.py

```diff
@@ -0,0 +1,429 @@
+"""Filtering module."""
+
+import inspect
+from functools import wraps
+from numbers import Number
+
+import numpy as np
+from scipy.signal import butter, sosfiltfilt
+
+from .. import core as nap
+
+
+def _get_butter_coefficients(cutoff, filter_type, sampling_frequency, order=4):
+    return butter(order, cutoff, btype=filter_type, fs=sampling_frequency, output="sos")
+
+
+def _compute_butterworth_filter(
+    data, cutoff, sampling_frequency=None, filter_type="bandpass", order=4
+):
+    """
+    Apply a Butterworth filter to the provided signal.
+    """
+    sos = _get_butter_coefficients(cutoff, filter_type, sampling_frequency, order)
+    out = np.zeros_like(data.d)
+    for ep in data.time_support:
+        slc = data.get_slice(start=ep.start[0], end=ep.end[0])
+        out[slc] = sosfiltfilt(sos, data.d[slc], axis=0)
+
+    kwargs = dict(t=data.t, d=out, time_support=data.time_support)
+    if isinstance(data, nap.TsdFrame):
+        kwargs["columns"] = data.columns
+    return data.__class__(**kwargs)
+
+
+def _compute_spectral_inversion(kernel):
+    """
+    Compute the spectral inversion.
+    Parameters
+    ----------
+    kernel: ndarray
+
+    Returns
+    -------
+    ndarray
+    """
```

Comment:
```suggestion
    """
    if len(kernel) % 2 == 0:
        raise ValueError("Spectral inversion requires an odd filter length.")
```

I know this is a private function, but I think a check here would make it the code more robust. If we edit our functions in the future, this is a good reminder